compiler/front/options.nim

```diff
@@ -665,7 +665,10 @@ func isEnabled*(conf: ConfigRef, report: ReportKind): bool =
           result = true
 
         of repTraceKinds:
-          result = true
+          # Semantic trace kinds are enabled by default and probably should
+          # not be changed - but these reports might (in theory) be
+          # modified at runtime.
+          result = report in conf.notes
```

Comment:
Could you list this (small functional) change in the commit body?

Reply:
Sorry, I just noticed your review as well (together with a notification about PR merge), but this is not a functional change, just part of the reporting enum sets was tangled together in a weird manner and there were two bugs cancelling each other out ((1)some reports were never added to the allowed `notes` and (2) for trace reports writing was always allowed).

Reply:
Added comment just documents the logic behind the check should have been added there several months ago

Reply:
Thank you for clarifying.